### PR TITLE
Add configuration for gcloud + self-host test

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This will supercede #109.

### CODE REVIEW

**Comment by acmiyaguchi** [comment]
The `generate-minio-configuration` script generates a file that looks like this:

```json
{
  "buckets": {
    "private": "b-private-032fc7c2cca96ddb",
    "shared": "b-shared-032fc7c2cca96ddb",
    "ingest": "b-ingest-032fc7c2cca96ddb"
  },
  "policy": {
    "internal": {
      "Version": "2012-10-17",
      "Statement": [
        {
          "Action": [
            "s3:*"
          ],
          "Effect": "Allow",
          "Resource": [
            "arn:aws:s3:::b-private-032fc7c2cca96ddb/*",
            "arn:aws:s3:::b-shared-032fc7c2cca96ddb/*",
            "arn:aws:s3:::b-ingest-032fc7c2cca96ddb/*"
          ],
          "Sid": ""
        }
      ]
    },
    "external": {
      "Version": "2012-10-17",
      "Statement": [
        {
          "Action": [
            "s3:*"
          ],
          "Effect": "Allow",
          "Resource": [
            "arn:aws:s3:::b-shared-032fc7c2cca96ddb/*"
          ],
          "Sid": ""
        }
      ]
    },
    "ingest": {
      "Version": "2012-10-17",
      "Statement": [
        {
          "Action": [
            "s3:*"
          ],
          "Effect": "Allow",
          "Resource": [
            "arn:aws:s3:::b-ingest-032fc7c2cca96ddb/*"
          ],
          "Sid": ""
        }
      ]
    }
  },
  "keys": {
    "internal": {
      "access_key": "b-032fc7c2cca96ddb",
      "secret_key": "a6b4094711a66498c561a8763b97d79087a989ab173ce84963c2247e7039fd74"
    },
    "external": {
      "access_key": "a-032fc7c2cca96ddb",
      "secret_key": "d3fe697dd6d4acc606d9a637e339ea5f04afeef55db14d8c4ad5321b31ae405f"
    },
    "ingest": {
      "access_key": "ingest-032fc7c2cca96ddb",
      "secret_key": "78aa4bb15f7571492472112dacb39a9760c5f4e46aba146813c6b5722478d81f"
    }
  }
}
```

I'm conflicted on how I should approach the set of scripts to make it easy coordinate on the particular variables. I'd still like to have most of the variables autopopulated into templates, but then have a few commands that can be used to generate fragments that can be copied into the dotenv files.

**Comment by acmiyaguchi** [comment]
Alongside testing with a local instance, I ran the test on a remote VM for server b. I copied the service account to the instance and changed `GOOGLE_APPLICATION_CREDENTIALS`. I changed the exposed port for the MinIO service from 9003 to 80 and made http available to the instance. I replaced `BUCKET_EXTERNAL_ENDPOINT` for the other configuration to the ip address of the VM and modified the script to remove the external network stuff.


<details>

```diff
From 0a871f3d476b7a523014495771ddc75d655ecb28 Mon Sep 17 00:00:00 2001
From: Anthony Miyaguchi <amiyaguchi@mozilla.com>
Date: Wed, 12 May 2021 17:00:49 -0700
Subject: [PATCH] Add modified configuration for hosted

---
 .../compose/ingest/docker-compose.yml               |  7 -------
 .../compose/server-a/docker-compose.yml             |  9 ---------
 deployment/testing-v4-gcloud-self/scripts/integrate | 13 -------------
 3 files changed, 29 deletions(-)

diff --git a/deployment/testing-v4-gcloud-self/compose/ingest/docker-compose.yml b/deployment/testing-v4-gcloud-self/compose/ingest/docker-compose.yml
index 8f002e4..4285734 100644
--- a/deployment/testing-v4-gcloud-self/compose/ingest/docker-compose.yml
+++ b/deployment/testing-v4-gcloud-self/compose/ingest/docker-compose.yml
@@ -1,9 +1,5 @@
 version: "3.9"
 
-networks:
-  testing-v4-gcloud-self-b_default:
-    external: true
-
 services:
   # https://docs.min.io/docs/minio-gateway-for-gcs.html
   gcs-gateway-ingest:
@@ -25,9 +21,6 @@ services:
       - ${GOOGLE_APPLICATION_CREDENTIALS?"missing credentials"}:/tmp/.credentials
       - ../../content.json:/app/config/content.json
       - ../../../../bin:/app/bin
-    networks:
-      - default
-      - testing-v4-gcloud-self-b_default
     depends_on:
       - gcs-gateway-ingest
     environment:
diff --git a/deployment/testing-v4-gcloud-self/compose/server-a/docker-compose.yml b/deployment/testing-v4-gcloud-self/compose/server-a/docker-compose.yml
index 345fba3..dc6e34c 100644
--- a/deployment/testing-v4-gcloud-self/compose/server-a/docker-compose.yml
+++ b/deployment/testing-v4-gcloud-self/compose/server-a/docker-compose.yml
@@ -1,11 +1,5 @@
 version: "3.9"
 
-# Only used when testing locally. When applied to an external host for server b,
-# use the public ip address of the node and remove this network dependency.
-networks:
-  testing-v4-gcloud-self-b_default:
-    external: true
-
 services:
   # https://docs.min.io/docs/minio-gateway-for-gcs.html
   gcs-gateway-a:
@@ -27,9 +21,6 @@ services:
     volumes:
       - ../../content.json:/app/config/content.json
       - ../../../../bin:/app/bin
-    networks:
-      - default
-      - testing-v4-gcloud-self-b_default
     depends_on:
       - gcs-gateway-a
     environment:
diff --git a/deployment/testing-v4-gcloud-self/scripts/integrate b/deployment/testing-v4-gcloud-self/scripts/integrate
index 671542c..e16260c 100755
--- a/deployment/testing-v4-gcloud-self/scripts/integrate
+++ b/deployment/testing-v4-gcloud-self/scripts/integrate
@@ -8,15 +8,6 @@ set -euo pipefail
 
 cd "$(dirname "$0")/.."
 
-# Start server B with minio server
-pushd compose/server-b
-docker-compose run --rm app bin/process &
-server_b_pid=$!
-popd
-
-# wait for the network to come online
-sleep 5
-
 # Copy data into the appropriate buckets
 pushd compose/ingest
 docker-compose run --rm app bin/generate
@@ -33,7 +24,6 @@ popd
 # `wait` is a blocking function, a failure in server B will not be detected
 # until timeout in server A.
 wait $server_a_pid
-wait $server_b_pid
 
 # clean up the containers
 pushd compose/server-a
@@ -45,6 +35,3 @@ docker-compose run --rm app bin/insert
 docker-compose down
 popd
 
-pushd compose/server-b
-docker-compose down
-popd
-- 
2.30.1
```
</details>

Everything works as expected.

File: deployment/testing-v4-gcloud-self/README.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
+# Testing configuration for v4 containers
+
+This directory contains terraform configuration to bring relevant resources for
+an integration test of the prio-processor v4.x containers.
+
+To create a new project that uses the same configuration, change the terraform
+backend appropriately. Here, the state is placed into a storage bucket that has
+been created beforehand. Ensure the project has also been created. Then:
```
Comment (wlach) [inline deployment/testing-v4-gcloud-self/README.md:8]: I think you could outline the steps here more clearly. I assume you mean a GCP project? And that it should be created before the storage bucket? Or is the storage bucket created by terraform?

File: deployment/testing-v4-gcloud-self/scripts/cleanup [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# Delegate cleanup of buckets to the appropriate service account
```
Comment (wlach) [inline deployment/testing-v4-gcloud-self/scripts/cleanup:3]: Same, it's not clear to me what this comment means.

File: deployment/testing-v4-gcloud-self/scripts/build [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# Delegate docker-compose build to each service
```
Comment (wlach) [inline deployment/testing-v4-gcloud-self/scripts/build:3]: Can you be more clear about what you mean by this comment?

File: deployment/testing-v4-gcloud-self/scripts/cleanup [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+set -euo pipefail
+cd "$(dirname "$0")/.."
+
+pushd compose/server-b
+docker-compose run --rm app bin/cleanup
+# keep the container around...
+
+pushd ../../compose/server-a
```
Comment (wlach) [inline deployment/testing-v4-gcloud-self/scripts/cleanup:12]: Maybe slightly clearly to use popd here, as you do elsewhere? Also, what does the comment `# keep the container around...` mean?

```suggestion
docker-compose run --rm app bin/cleanup
popd
# keep the container around...

pushd compose/server-a
```

**Review by wlach — APPROVED** [review]
Looks pretty fine to me, some very minor nits around documentation and comments.

**Pull request merged**